src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,174 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            setFirstValue(value);
+            return;
+        }
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        addCheck(index);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        linkBefore(value, findNodeByIndex(index));
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            return;
+        }
+        for (T item : list) {
+            this.add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> oldNode = findNodeByIndex(index);
+        T oldItem = oldNode.item;
+        oldNode.item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> node = findNodeByIndex(index);
+        T value = node.item;
+        unlink(node);
+        return value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> node = getNode(object);
+        if (node == null) {
+            return false;
+        }
+        unlink(node);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void addCheck(int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Your index was " + index
+                    + ", but size is " + size);
+        }
+    }
+
+    private void indexCheck(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Your index was " + index
+                    + ", but size is " + size);
+        }
+    }
+
+    private Node<T> getNode(T value) {
+        Node<T> node = head;
+        for (int i = 0; i < size; i++) {
+            if ((node.item == value) || (node.item != null && node.item.equals(value))) {
+                return node;
```

Comment:
instead of return node and make some additional check, you can call method unlink and after this return true here, otherwise return false. It will optimize your code